sycl/CMakeLists.txt

```diff
@@ -383,6 +383,12 @@ if(OpenCL_INSTALL_KHRONOS_ICD_LOADER AND TARGET OpenCL-ICD)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS OpenCL-ICD)
 endif()
 
+# Build and install lld as part of the sycl-toolchain if available
+if(TARGET lld)
```

Comment:
I hope it works.
I thought about examining LLVM_ENABLE_PROJECTS value, but assuming that lld's project cmake files are parsed before this cmake file, this should work as well.

Reply:
Good point, I was under the incorrect assumptions that the target would need to already be defined for `add_dependencies` to work anyway, but it looks like `add_dependencies` still works even if the target is defined later, so I've replaced the conditions with your initial suggestion.